### ISSUE TITLE
LevelDB error when running Tribler on Windows

### ISSUE THREAD

**Issue by lfdversluis** [Tribler/tribler#1837]
I am running Tribler with the corrupt .Tribler folder of @theodoor to fix #1724 yet I run into a old, but slightly different issue (looks like #1535)

```
ERROR   1451478209.16 SqliteCacheDBHandler:364   Could not create a TorrentDef i
nstance 'Tl\xf1_rM\x19\xc41\x9c\xc1{\x17\x9d~\x03_\x89\xc1\xf4' 1435581520 u'ubu
ntu-14.04.2-desktop-amd64.iso' ((u'ubuntu-14.04.2-desktop-amd64.iso', 1044381696
),) ('http://torrent.ubuntu.com:6969/announce', 'http://ipv6.torrent.ubuntu.com:
6969/announce') {'dispersy_id': 24973}
Traceback (most recent call last):
  File "C:\Users\Administrator\Documents\TriblerFork\Tribler\Core\CacheDB\Sqlite
CacheDBHandler.py", line 355, in addExternalTorrentNoDef
    torrent_id = self._addTorrentToDB(torrentdef, extra_info)
  File "C:\Users\Administrator\Documents\TriblerFork\Tribler\Core\CacheDB\Sqlite
CacheDBHandler.py", line 442, in _addTorrentToDB
    self._addTorrentTracker(torrent_id, torrentdef, extra_info)
  File "C:\Users\Administrator\Documents\TriblerFork\Tribler\Core\CacheDB\Sqlite
CacheDBHandler.py", line 514, in _addTorrentTracker
    self.addTorrentTrackerMappingInBatch(torrent_id, list(new_tracker_set))
  File "C:\Users\Administrator\Documents\TriblerFork\Tribler\Core\CacheDB\Sqlite
CacheDBHandler.py", line 672, in addTorrentTrackerMappingInBatch
    if infohash and self.session.has_collected_torrent(infohash):
  File "C:\Users\Administrator\Documents\TriblerFork\Tribler\Core\Session.py", l
ine 568, in has_collected_torrent
    return hexlify(infohash) in self.lm.torrent_store
  File "C:\Users\Administrator\Documents\TriblerFork\Tribler\Core\leveldbstore.p
y", line 107, in __contains__
    self.__getitem__(key)
  File "C:\Users\Administrator\Documents\TriblerFork\Tribler\Core\leveldbstore.p
y", line 86, in __getitem__
    return self._db.Get(key)
LevelDBError: IO error: collected_torrents\000005.sst: Could not create random a
ccess file.
```

**Comment by lfdversluis** [comment]
Related: https://github.com/fusesource/leveldbjni/issues/28 and apparently known here: https://jira.duraspace.org/browse/FCREPO-1743 yet no fix.

**Comment by whirm** [comment]
According to fusesource/leveldbjni#28: 
`Retrying to read after an exception is thrown (which I filtered if the exception message contains error code 32 :S ) seems to fix the problem.`

Do you want to create a patch to implement this workaround? (Also document why this is done and a reference to the tickets you can find)

**Comment by lfdversluis** [comment]
@whirm If you continue reading you see that the problem persists, even after his second work around/fix

**Comment by whirm** [comment]
@lfdversluis yes, but it seems to be due to a different reason (iterator not destroyed) hopefully we won't trigger this issue.

**Comment by lfdversluis** [comment]
A simple try: except: doesn't work unfortunately.

**Comment by whirm** [comment]
you are saying the `try` is not catching the exception?

**Comment by lfdversluis** [comment]
oh it is, but if I do the call again in the except part it just throws another exception.

**Comment by whirm** [comment]
I'd reschedule the flush to happen a bit later to give levelDB the opportunity to finish the leveling.

**Comment by lfdversluis** [comment]
@whirm does not work. I've compiled a newer version of leveldb today but that also does not solve the issue. Creating a very minimal program also does not seem to fix the issue.

**Comment by devos50** [comment]
Fixed in #1853
